x-pack/legacy/plugins/siem/public/containers/hosts/first_last_seen/index.ts

```diff
@@ -51,31 +51,33 @@ export function useFirstLastSeenHostQuery<TCache = object>(
           hostName,
           defaultIndex: chrome.getUiSettingsClient().get(DEFAULT_INDEX_KEY),
         },
+        context: {
+          fetchOptions: {
+            signal,
+          },
+        },
       })
       .then(
         result => {
           updateLoading(false);
           updateFirstSeen(get('data.source.HostFirstLastSeen.firstSeen', result));
           updateLastSeen(get('data.source.HostFirstLastSeen.lastSeen', result));
           updateErrorMessage(null);
-          return result;
         },
         error => {
           updateLoading(false);
+          updateFirstSeen(null);
+          updateLastSeen(null);
           updateErrorMessage(error.message);
-          return error;
         }
       );
   }
 
   useEffect(() => {
-    try {
-      fetchFirstLastSeenHost();
-    } catch (err) {
-      updateFirstSeen(null);
-      updateLastSeen(null);
-      updateErrorMessage(err.toString());
-    }
+    const abortCtrl = new AbortController();
```

Comment:
Did not know about this -- very cool!

Reply:
Same, that's pretty neat and fetch takes signals too and does cancels.

Reply:
Also it's used throughout the `kibana` code base so it looks to be polyfilled and a great thing to use.